packages/core/src/server/devServer.ts

```diff
@@ -197,11 +220,26 @@ export async function createDevServer<
     compileMiddlewareAPI,
   });
 
+  const environmentAPI = Object.fromEntries(
+    Object.keys(options.context.environments).map((name, index) => {
+      return [
+        name,
+        {
+          getStats: async () => {
+            await waitFirstCompileDone;
+            return _stats[index];
```

Comment:
Should we throw an error when `runCompile` is false? The return type is not stats